feat: revert navbar to single row

surajsharma opened the pull request:
eta: 4 hours
#493

markholdex commented:
@predrag-us please have a design review of this issue.

predrag-us commented:
@surajsharma Please compare to the design.

zolotokrylin commented:
@predrag-us @markholdex next time please use "review" feature, and you strongly don't agree with the solution provided, use "Request changes" option. 

We are tracking this events.

predrag-us reviewed (changes requested):
@surajsharma Please have a good look at the design, the implementation is way off.

surajsharma commented:
@predrag-us i used pixelperfect to ensure it is per design, kindly share screenshots or description for where/what exactly is not per design

surajsharma commented:
@predrag-us i think you must be taking about the new design, which is different pr for #493, in this pr i  have reverted to original state (single column). next, i will be updating to the new design

predrag-us commented:
@surajsharma 

The new design is 99% the same as the old, I've only made minor tweaks so it can work on all breakpoints and these changes are focused on the middle section, meaning that the left and right sections are the same. Therefore, this is not how the navigation looked prior to the changes. There are differences I won't describe since you should be able to notice them.  

My advice is to implement the navigation as is in the design since there's no sense to me to do a new PR for a 1% difference.

surajsharma commented:
@predrag-us 

> @surajsharma

> The new design is 99% the same as the old

the remaining 1% requires code functionality, hence the separate pr

> My advice is to implement the navigation as is in the design since there's no sense to me to do a new PR for a 1% difference.

am a bit uncomfortable mixing new code with a clearly separately marked AC 

pls advise @markholdex @georgeciubotaru

georgeciubotaru commented:
@surajsharma Please leave a log with estimation.

Let's wrap up this PR ASAP, minimize the scope so we can merge it without breaking any logic, after that you can continue working in another PR

surajsharma commented:
> @surajsharma Please leave a log with estimation.

@georgeciubotaru need a couple hours here, assuming redesign does not include new components i can see in the design like the search box..pls confirm

predrag-us commented:
@surajsharma No, we don't need the search bar and other elements we don't currently have.

georgeciubotaru commented:
@surajsharma Minimize the scope then

Adjust this PR to not break anything on prod, and let's merge it (max 1h), after then you can split the Problem into smaller ones.

surajsharma commented:
@predrag-us @georgeciubotaru made the necessary adjustments here, kindly have a look

predrag-us reviewed (changes requested):
@surajsharma Please notice that the middle section of the navigation is centered on larger screens. Also, there are two variants ⏤ Default and Sticky. Default should be displayed instantly while Sticky should be displayed on scroll. Lastly, check the mobile navigation, since it's broken and not by design (screenshot).

<img width="443" alt="CleanShot 2023-08-25 at 13 47 55@2x" src="https://github.com/holdex/holdex-venture-studio/assets/18451941/d6933535-1bd8-46b9-8ea8-cf791173399e">

georgeciubotaru commented on src/lib/styles/variables.scss:
```diff
@@ -23,6 +23,7 @@ $light-colors: (
   t2: toSpacedRgb(#2f323d),
   t3: toSpacedRgb(#5d657a),
   t4: toSpacedRgb(#ffffff),
+  i1: toSpacedRgb(#10141f),
```
@surajsharma why do you add new colors. we already have that color

surajsharma replied:
removed

georgeciubotaru commented on src/lib/styles/variables.scss:
```diff
@@ -40,6 +41,7 @@ $dark-colors: (
   t2: toSpacedRgb(#d7dae0),
   t3: toSpacedRgb(#a2a7b8),
   t4: toSpacedRgb(#10141f),
```
same here

surajsharma replied:
removed

georgeciubotaru commented on tailwind.config.cjs:
```diff
@@ -75,6 +75,7 @@ module.exports = {
       t2: 'rgb(var(--t2) / <alpha-value>)',
       t3: 'rgb(var(--t3) / <alpha-value>)',
       t4: 'rgb(var(--t4) / <alpha-value>)',
+      i1: 'rgb(var(--i1) / <alpha-value>)',
```
same here

georgeciubotaru reviewed (changes requested):
see my first review comments

predrag-us reviewed (changes requested):
@surajsharma 

1. The middle content isn't really centered. You have to make the logo wrapper and the action (buttons) wrapper the same width in order for the middle content to be centered.
2. Mobile view is still not by design. Please notice that the action (buttons) aren't always visible. The height of the navigation is also incorrect.
3. Please notice that the fade goes over the logo, that shouldn't be the case. 
4. Please check my previous review again and the point about Default and Sticky variants. 
5. The navigation is unusable between 820px and 480px. Please notice the Screen : Middle variant. It should have the same behavior as mobile (horizontally scrollable, with fade).

surajsharma commented:
> @surajsharma
> 
> 1. The middle content isn't really centered. You have to make the logo wrapper and the action (buttons) wrapper the same width in order for the middle content to be centered.
> 2. Mobile view is still not by design. Please notice that the action (buttons) aren't always visible. The height of the navigation is also incorrect.
> 3. Please notice that the fade goes over the logo, that shouldn't be the case.
> 4. Please check my previous review again and the point about Default and Sticky variants.
> 5. The navigation is unusable between 820px and 480px. Please notice the Screen : Middle variant. It should have the same behavior as mobile (horizontally scrollable, with fade).

@predrag-us  i have addressed all these concerns, please have a look

predrag-us reviewed (changes requested):
@surajsharma 

1. What happened to the middle section (content is separated now)? Please check this design and revert to the correct version. 
2. Mobile view is still not by design. Please notice that the action (buttons) aren't always visible but only when you scroll to the very end. Please notice that the active indicator shouldn't be above the line (screenshot). 
3. Please check my previous review again and the point about Default and Sticky variants.

<img width="70" alt="CleanShot 2023-08-30 at 10 28 41@2x" src="https://github.com/holdex/holdex-venture-studio/assets/18451941/1c373898-6e6c-474e-80fa-a022b9f196fa">

surajsharma commented:
@predrag-us  thanks!

i was not sure about 1. so i let it be, will correct it.

there was some problem in my understanding of 2. will fix now

> 3. Please check my previous review again and the point about Default and Sticky variants.

however, i do not still understand this, or this

> Default should be displayed instantly while Sticky should be displayed on scroll.

do you not see the default one instantly and the sticky one on scroll?

predrag-us commented:
@surajsharma 

> do you not see the default one instantly and the sticky one on scroll? 

It's always Sticky. 

Please notice that you have two variants in the design, one without a background and a border and one with a background and a border. 

The Default variant should be displayed instantly, while the Sticky variant should be triggered on scroll.

markholdex commented:
@surajsharma can this PR be reviewed?

markholdex commented:
Visually seems like working fine on desktop. Will look on mobile for now

markholdex commented:
Mobile looks good too. Visually, it's ready for release.

markholdex commented:
@georgeciubotaru please review quickly or delegate review

surajsharma commented:
> @surajsharma
> 
> > do you not see the default one instantly and the sticky one on scroll?
> 
> It's always Sticky.

@predrag-us please leave a comment on figma explaining what if anything is off here

@markholdex i'm ok with this if @predrag-us is cool

georgeciubotaru commented on src/routes/(pages)/layout.pug:
```diff
@@ -34,23 +34,47 @@ mixin main-nav
+              class="w-8 h-8 "
+            )
+
+          div(id="secondary-navbar" use:scrollAction class!=" z-[98] bg-l1 overflow-x-scroll scrollbar-hide w-full")
```
@surajsharma let's not use so high values for `z-index`

In your case, the `navbar` has 98 index, but the button (theme switcher) has 20, so the button is not clickable

georgeciubotaru replied:
@surajsharma how about the `cta` button

`Partner with us` button should be transformed into a `fixed` button
@markholdex can you confirm that?

surajsharma replied:
@georgeciubotaru done on the `z-index` issue

georgeciubotaru commented:
@surajsharma Please see my comments, let's wrap up this task

surajsharma commented:
> Please notice that you have two variants in the design, one without a background and a border and one with a background and a border.

> The Default variant should be displayed instantly, while the Sticky variant should be triggered on scroll.

fix this to thedegree i could, note that bg is added where it was necessary and border is available `xs` and below

fixed every other issue as well

@georgeciubotaru @predrag-us

georgeciubotaru commented:
@surajsharma Please first push your changes, then request review

surajsharma commented:
> @surajsharma Please first push your changes, then request review

@georgeciubotaru sorry accidentally pushed to wrong branch

georgeciubotaru commented:
@surajsharma The previous version worked better, scroll should be enabled only for the nav items

Logo and theme switcher should remain in place

surajsharma commented:
@predrag-us i think i understood your concern about the sticky/normal variants, but not having a bg causes this problem 
<img width="1287" alt="image" src="https://github.com/holdex/holdex-venture-studio/assets/5298624/9bae6158-745d-4012-a5de-f93e62818f20">


so i am adding bg wherever necessary 

cc @georgeciubotaru

georgeciubotaru commented:
@surajsharma 

from this variant: [holdex-venture-studio-nm0hkr8em-holdex-accelerator.vercel.app](https://holdex-venture-studio-nm0hkr8em-holdex-accelerator.vercel.app/)

Just resolve the `z-index` issue and it's done

Your current version is broken

surajsharma commented:
@georgeciubotaru have fixed the z-index (max `z-[21]`) issue and achieved sameness with

> https://holdex-venture-studio-nm0hkr8em-holdex-accelerator.vercel.app/

georgeciubotaru commented:
@surajsharma The correct version from the URL I've provided
![image](https://github.com/holdex/holdex-venture-studio/assets/8217518/d37ad661-0fe9-4874-86c0-1c7a4a27abb5)


Your newest version
![image](https://github.com/holdex/holdex-venture-studio/assets/8217518/e5e0f18e-a776-4ae3-883e-9f3dcf21b3d6)


Please compare changes from this commit until now
https://github.com/holdex/holdex-venture-studio/pull/499/commits/d5c59cbfe28806506ed36ac734109cc1e81fc02d

surajsharma commented:
@georgeciubotaru ok how about now?
